core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -2219,6 +2207,42 @@ protected void convertFrom(
     }
   }
 
+  private List<String> getFieldAliases(SqlCall asCall) {
+    if (!validator.getConformance().allowAliasUnnestColumns()) {
+      return null;
+    }
+
+    List<String> columnAliases = new ArrayList<>();
+    for (SqlNode sqlNode : Util.skip(asCall.getOperandList(), 2)) {
+      columnAliases.add(((SqlIdentifier) sqlNode).getSimple());
+    }
+    return columnAliases;
+  }
+
+  private void convertUnnest(Blackboard bb, SqlCall from, List<String> fieldAliases) {
+    SqlCall call;
+    call = from;
+    final List<SqlNode> nodes = call.getOperandList();
+    final SqlUnnestOperator operator = (SqlUnnestOperator) call.getOperator();
+    for (SqlNode node : nodes) {
+      replaceSubQueries(bb, node, RelOptUtil.Logic.TRUE_FALSE_UNKNOWN);
+    }
+    final List<RexNode> exprs = new ArrayList<>();
+    final List<String> fieldNames = new ArrayList<>();
```

Comment:
you don't need two separate arrays. `RelBuilder.project` will apply aliases if you have used `RelBuilder.alias` to wrap expressions in 'as'. Use `project` rather than `projectNamed`.